tools/distpartitioning/parmetis_wrapper.py

```diff
@@ -0,0 +1,331 @@
+    <global_src_id> <global_dst_id>
+
+    Here `global` prefix means that globally unique identifier assigned each node
+    in the input graph. In this context globally unique means unique across all the
+    nodes in the input graph.
+
+    Parameters:
+    -----------
+    schema_map : json dictionary
+        dictionary created by reading the metadata.json file for the input dataset
+    output : string
+        location of storing the node-weights and edge files for ParMETIS
+    """
+    rank = get_proc_info()
+    type_nid_dict, ntype_gnid_offset = get_idranges(schema_map[constants.STR_NODE_TYPE],
+                                        schema_map[constants.STR_NUM_NODES_PER_CHUNK])
+
+    #Regenerate edge files here...▒
+    # in the following format....
+    edge_data = schema_map[constants.STR_EDGES]
+    etype_names = schema_map[constants.STR_EDGE_TYPE]
+    etype_name_idmap = {e : idx for idx, e in enumerate(etype_names)}
+    edge_tids, _ = get_idranges(schema_map[constants.STR_EDGE_TYPE],
+                                    schema_map[constants.STR_NUM_EDGES_PER_CHUNK])
+
+    outdir = Path(output)
+    os.makedirs(outdir, exist_ok=True)
+    edge_files = []
+    num_parts = len(schema_map[constants.STR_NUM_EDGES_PER_CHUNK][0])
+    for etype_name, etype_info in edge_data.items():
+
+        edge_info = etype_info[constants.STR_DATA]
+
+        #edgetype strings are in canonical format, src_node_type:edge_type:dst_node_type
+        tokens = etype_name.split(":")
+        assert len(tokens) == 3
+
+        src_ntype_name = tokens[0]
+        rel_name = tokens[1]
+        dst_ntype_name = tokens[2]
+
+        data_df = csv.read_csv(edge_info[rank],
+                      read_options=pyarrow.csv.ReadOptions(autogenerate_column_names=True),
+                      parse_options=pyarrow.csv.ParseOptions(delimiter=' '))
+        data_f0 = data_df['f0'].to_numpy()
+        data_f1 = data_df['f1'].to_numpy()
+
+        global_src_id = data_f0 + ntype_gnid_offset[src_ntype_name][0, 0]
+        global_dst_id = data_f1 + ntype_gnid_offset[dst_ntype_name][0, 0]
+        cols = [global_src_id, global_dst_id]
+        col_names = ["global_src_id", "global_dst_id"]
+        #global_type_eid = np.arange(edge_tids[etype_name][rank][0],\
+        #     edge_tids[etype_name][rank][1] ,dtype=np.int64)
+        #etype_id = etype_name_idmap[etype_name] * \
+        #        np.ones(shape=(data_df['f0'].to_numpy().shape), dtype=np.int64)
+
+        out_file = edge_info[rank].split("/")[-1]
+        out_file = os.path.join(outdir, 'edges_{}'.format(out_file))
+        options = csv.WriteOptions(include_header=False, delimiter=" ")
+        options.delimiter = " "
+
+        csv.write_csv(pyarrow.Table.from_arrays(cols, names=col_names), out_file, options)
+
+        edge_files.append(out_file)
+
+    return edge_files
+
+def read_node_features(schema_map, tgt_ntype_name, feat_names):
```

Comment:
Why need to read node features during partitioning?

Reply:
node features (train_mask, test_mask and valid_mask) are read and will form the node constraints. These constraints will be used to create node weights which will be used by ParMETIS for partitioning purposes.